src/ioctls/vcpu.rs

```diff
@@ -1210,16 +1214,27 @@ impl VcpuFd {
     /// let vm = kvm.create_vm().unwrap();
     /// let vcpu = vm.create_vcpu(0).unwrap();
     ///
-    /// // KVM_GET_REG_LIST demands that the vcpus be initalized.
-    /// let mut kvi: kvm_bindings::kvm_vcpu_init = kvm_bindings::kvm_vcpu_init::default();
-    /// vm.get_preferred_target(&mut kvi).unwrap();
-    /// vcpu.vcpu_init(&kvi).expect("Cannot initialize vcpu");
+    /// // KVM_GET_REG_LIST on Aarch64 demands that the vcpus be initalized.
+    /// #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    /// {
+    ///     let mut kvi: kvm_bindings::kvm_vcpu_init = kvm_bindings::kvm_vcpu_init::default();
+    ///     vm.get_preferred_target(&mut kvi).unwrap();
+    ///     vcpu.vcpu_init(&kvi).expect("Cannot initialize vcpu");
+    /// }
+    ///
+    /// let nr_regs = if cfg!(any(target_arch = "arm", target_arch = "aarch64")) {
+    ///     500
+    /// } else if cfg!(target_arch = "riscv64") {
+    ///     200
+    /// } else {
+    ///     unreachable!();
+    /// };
```

Comment:
Here I also think it's fine to have the code example only for ARM, with the riscv case moved to unittests (or vice-versa). The idea of the doc example is more to give an idea of how to use the API, and since the API itself is identical on multiple architectures, I don't really think we need to have the same code twice, just with different constants